File: melbing-app/app/Models/ShipLog.php
<?php

namespace App\Models;

use Carbon\Carbon;
use Database\Factories\ShipLogFactory;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ShipLog extends Model
{
    /** @use HasFactory<ShipLogFactory> */
    use HasFactory;

    protected $fillable = [
        'unix_time',
        'humidity',
        'inside_temp',
        'outside_temp',
        'water_temp',
        'refrigerator_temp',
        'pressure',
        'consumer_bat_v',
        'start_bat_v',
        'shore_power',
        'wind_speed',
        'relative_wind_angle',
        'true_wind_angle',
        'heating_element',
        'dehumidifier',
        'battery_charger',
        'solar_charger',
        'daylight_saving',
        'timezone',
    ];

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'unix_time' => 'integer',
            'humidity' => 'float',
            'inside_temp' => 'float',
            'outside_temp' => 'float',
            'water_temp' => 'float',
            'refrigerator_temp' => 'float',
            'pressure' => 'float',
            'consumer_bat_v' => 'float',
            'start_bat_v' => 'float',
            'shore_power' => 'boolean',
            'wind_speed' => 'float',
            'relative_wind_angle' => 'float',
            'true_wind_angle' => 'float',
            'heating_element' => 'boolean',
            'dehumidifier' => 'boolean',
            'battery_charger' => 'boolean',
            'solar_charger' => 'boolean',
            'daylight_saving' => 'boolean',
        ];
    }

    public function getRecordedAtAttribute(): Carbon
    {
        return Carbon::createFromTimestamp($this->unix_time);
    }
}

File: melbing-csharp/src/Melbing.ShipLog.Domain/Entities/ShipLog.cs
namespace Melbing.ShipLog.Domain.Entities;

public class ShipLog
{
    public long Id { get; set; }

    public long UnixTime { get; set; }

    public float? Humidity { get; set; }

    public float? InsideTemp { get; set; }

    public float? OutsideTemp { get; set; }

    public float? WaterTemp { get; set; }

    public float? RefrigeratorTemp { get; set; }

    public float? Pressure { get; set; }

    public float? ConsumerBatV { get; set; }

    public float? StartBatV { get; set; }

    public bool ShorePower { get; set; }

    public float? WindSpeed { get; set; }

    public float? RelativeWindAngle { get; set; }

    public float? TrueWindAngle { get; set; }

    public bool HeatingElement { get; set; }

    public bool Dehumidifier { get; set; }

    public bool BatteryCharger { get; set; }

    public bool SolarCharger { get; set; }

    public bool DaylightSaving { get; set; }

    public string? Timezone { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public DateTimeOffset RecordedAt => DateTimeOffset.FromUnixTimeSeconds(UnixTime);
}
